buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/NamingConventionsTask.groovy

```diff
@@ -57,8 +57,25 @@ public class NamingConventionsTask extends LoggedExec {
     @Input
     boolean skipIntegTestInDisguise = false
 
+    /**
+     * Superclass for all tests.
+     */
+    @Input
+    String testClass = 'org.apache.lucene.util.LuceneTestCase'
+
+    /**
+     * Superclass for all integration tests.
+     */
+    @Input
+    String integTestClass = 'org.elasticsearch.test.ESIntegTestCase'
+
     public NamingConventionsTask() {
-        dependsOn(classpath)
+        // Extra classpath contains the actual test
+        FileCollection extraClasspath = project.findProject(':build-tools').sourceSets.main.output
```

Comment:
This will not work when used with a plugin outside ES source...I think what you want is to add a configuration used by this task, and add build-tools as the only dep in that configuration. Then resolve the configuration at runtime to get the jar to use.